splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/network/NetworkChangeMonitorBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.rum.internal.instrumentation.network;
+
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A builder of {@link NetworkChangeMonitor}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class NetworkChangeMonitorBuilder {
+
+    final CurrentNetworkProvider currentNetworkProvider;
+    final List<AttributesExtractor<CurrentNetwork, Void>> additionalExtractors = new ArrayList<>();
+
+    NetworkChangeMonitorBuilder(CurrentNetworkProvider currentNetworkProvider) {
+        this.currentNetworkProvider = currentNetworkProvider;
+    }
+
+    /** Adds an {@link AttributesExtractor} that will extract additional attributes. */
+    public NetworkChangeMonitorBuilder addAttributesExtractor(
+            AttributesExtractor<CurrentNetwork, Void> extractor) {
+        additionalExtractors.add(extractor);
+        return this;
+    }
+
+    /**
+     * Returns a new {@link NetworkChangeMonitor} with the settings of this {@link
+     * NetworkChangeMonitorBuilder}.
+     */
+    public NetworkChangeMonitor build() {
```

Comment:
I think this is conventional enough to not need explanation.

```suggestion
    public NetworkChangeMonitor build() {
```